### ISSUE TITLE
Grid creation with large max_grid_size

### ISSUE THREAD

**Issue by NeilZaim** [ECP-WarpX/WarpX#1206]
Not sure if this is an issue with amrex or warpx but I'm putting it here. :)

We wanted to run a series a simulations ensuring that we always have 1 box per MPI/GPU regardless of the size/parameters of the simulation. From what I (thought I) understood of amrex grid creation, I thought that we would be able to achieve this by setting a low value for `blocking_factor` (typically 8) and a ridiculously high value for `max_grid_size` (something like 2^25). (That's what we wanted to put by default in warpx to make it more user friendly no?)

But it turns out that if `max_grid_size` is too large the number of boxes in the domain decomposition decreases so that we have less boxes than MPI processes (eventually if `max_grid_size` is very large we end up with no domain decomposition at all so one process will do all the work).

I've poked around in the code and from the following line in amrex it looks like the grids will never be smaller than `max_grid_size / 8`

https://github.com/AMReX-Codes/amrex/blob/fb413521ab2ebd9fa7614c1434a4c913a3516b06/Src/AmrCore/AMReX_AmrMesh.cpp#L423

If I replace it with something like `for (int cnt = 1; cnt <= 64; cnt *= 2)` the behavior becomes closer to what I expected.

Is this normal? And if so, is there a clean way to ensure that we always have exactly 1 box per MPI/GPU?

Thanks!

PS: I've also noticed that we can have `max_grid_size` not divisible by `blocking_factor`, which I thought was not possible (although the grid creation is a bit strange in this case).

**Comment by WeiqunZhang** [comment]
Yes, you are right that `AmrMesh::ChopGrids` cannot deal with very large `max_grid_size`.  We could modify that function to handle that.  But it will not guarantee exactly 1 Box per process.
